Vehicle.h

```diff
@@ -3,12 +3,14 @@
 #include <string>
 #include <iostream>
 
+struct HighwayPatrol;
+
 struct Vehicle
 {
     Vehicle(const std::string& n) : name(n) { }
     
-    virtual ~Vehicle() = default;
-    Vehicle(const Vehicle&) = default;
+    virtual ~Vehicle();// = default;
+    Vehicle(const Vehicle&);// = default;
     Vehicle& operator=(const Vehicle&) = default;
```

Comment:
move this copy-assign operator implementation to the cpp file, with the other implementations.
- [ ] revise